TypeCobol/Compiler/Scanner/Token.cs

```diff
@@ -405,10 +406,48 @@ public bool CompareForReplace(Token? comparisonToken)
                 return false;
             }
 
+            /*
+             Replace ==C-Nb==   by ==xxx==
+             X(C-Nb)  -> Ok because parenthesis around
+             X(C-NbX) -> Ko because followed by X
+
+             Replace ==C-Nb)=== by ==xxx==
+             X(C-Nb)  -> Ok because parenthesis before and end of token after
+             X(C-NbX) -> Ko IndexOf don't match
+
+             Replace ==C-Nb==   by ==xxx==
+             X:C-Nb:  -> Ok because semicolon around
+             X:C-NbX: -> Ko because followed by X
+
+             Replace ==C-Nb)==  by ==xxx==
+             X:C-Nb:  -> Ko IndexOf don't match
+             X:C-NbX: -> Ko IndexOf don't match
+
+             Replace ==C-Nb:==  by ==xxx==
+             X:C-Nb:  -> Ok because semicolon before and end of token after
+             X:C-NbX: -> Ko IndexOf don't match
+             */
+
             //PartialCobolWord and PictureCharacterString are text based (and must be rescanned later as a whole)
             if (TokenType == TokenType.PartialCobolWord || TokenType == TokenType.PictureCharacterString)
             {
-                return NormalizedText.IndexOf(comparisonToken.NormalizedText, StringComparison.OrdinalIgnoreCase) >= 0;
+                var startIndexFound = NormalizedText.IndexOf(comparisonToken.NormalizedText, StringComparison.OrdinalIgnoreCase);
```

Comment:
Store `NormalizedText` and `comparisonToken.NormalizedText` into local variables to avoid multiple property calls.

Reply:
Done